src/prefect/logging/handlers.py

```diff
@@ -355,3 +359,40 @@ def close(self) -> None:
             worker.flush()
 
         return super().close()
+
+
+class PrefectConsoleHandler(logging.Handler):
+    def __init__(
+        self,
+        highlighter: Highlighter = PrefectConsoleHighlighter,
+        styles: Dict[str, str] = None,
+        level: Union[int, str] = logging.NOTSET,
+        console: Optional[Console] = None,
+    ):
+        if console is None:
+            styled_console = PREFECT_LOGGING_STYLED_CONSOLE.value()
+            if styled_console:
+                highlighter = highlighter()
+                theme = Theme(styles, inherit=False)
+            else:
+                highlighter = NullHighlighter()
+                theme = Theme(inherit=False)
+            console = Console(highlighter=highlighter, theme=theme)
+        self.console = console
+        super().__init__(level=level)
+
+    def emit(self, record: logging.LogRecord):
+        try:
+            record.message = record.getMessage()
+            if self.formatter:
+                formatter = self.formatter
+                if hasattr(formatter, "usesTime") and formatter.usesTime():
+                    record.asctime = formatter.formatTime(record, formatter.datefmt)
+                message = formatter.formatMessage(record)
+            else:
+                message = record.message
+            self.console.print(message)
+        except RecursionError:  # See issue 36272
```

Comment:
This is a rich issue? Can you include a full link and a short description of the issue?

Reply:
This is actually copied from `logging.StreamHandler.emit()`

I can add a note.